console_conf/controllers/identity.py

```diff
@@ -70,12 +70,15 @@ def get_device_owner():
     """ Get device owner, if any """
     con = SnapdConnection('', '/run/snapd.socket')
     for user in con.get('v2/users').json()['result']:
-        user = user['email'].split('@')[0]
-        if os.path.isdir('/home/' + user):
+        if 'username' not in user:
+            continue
+        username = user['username']
+        homedir = '/home/' + username
+        if os.path.isdir(homedir):
             return {
                 'username': user,
```

Comment:
won't `user` here be an object and not the string value of the username? I'm not super familiar with the rest of this code so perhaps it doesn't matter?

Reply:
Yes you're right, fixed!